synapse/tests/test_link.py

```diff
@@ -12,12 +12,17 @@
 
 from synapse.tests.common import *
 
+
 class FooBar:
 
-    def foo(self):
+    bigdata = os.urandom(1000 * 1000 * 50)
```

Comment:
a static large string would be quicker than calling os.urandom.

Reply:
It only reliably breaks at >= 50 MB and didn't want to commit a big junk file like that. It only takes about 1s to gen this data.

Reply:
We could also write a much more targeted test, this test basically uses brute force to trigger the behavior.

Reply:
I'd like to see a more targeted test that doesn't involve racing against the multiplexer.  😁

Reply:
I added more more targeted tests. I'll remove this once they all pass.